lib/peg.d.ts

```diff
@@ -0,0 +1,137 @@
+// Based on PEG.js Type Definitions by: vvakame <https://github.com/vvakame>, Tobias Kahlert <https://github.com/SrTobi>, C.J. Bell <https://github.com/siegebell>
+
+declare namespace PEG {
+  function parse(input: string): any;
+
+  interface Location {
+    line: number;
+    column: number;
+    offset: number;
+  }
+
+  interface LocationRange {
+    start: Location;
+    end: Location;
+  }
+
+  class SyntaxError {
+    line: number;
+    column: number;
+    offset: number;
+    location: LocationRange;
+    expected: any[];
+    found: any;
+    name: string;
+    message: string;
+  }
+}
+
+export type Location = PEG.Location;
+export type LocationRange = PEG.LocationRange;
+
+export interface ExpectedItem {
+  type: string;
+  value?: string;
+  description: string;
+}
+
+export interface PeggyError extends Error {
+  name: string;
+  message: string;
+  location: LocationRange;
+  found?: any;
+  expected?: ExpectedItem[];
+  stack?: any;
+}
+
+// for backwards compatibility with PEGjs
+export interface PegjsError extends PeggyError {}
```

Comment:
How do we see the backward-compatibility bits being used?

Reply:
I think it's conceivable that someone is doing `import { PegjsError } from "pegjs"` so didn't want to just rename it, and this was the only type that includes the old name.

Reply:
I will change it to just a type alias though, no need to extend